pinot-core/src/main/java/com/linkedin/pinot/core/startree/hll/HllConfig.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.startree.hll;
+
+import com.google.common.base.Preconditions;
+
+import java.util.*;
+
+/**
+ * HllConfig is the config for hll index generation.
+ * Note it is only effective when StarTree index generation is enabled.
+ */
+public class HllConfig {
+
+    private int hllLog2m;
+    private Map<String, String> hllDeriveFieldNameMap;
+    private boolean enableHllIndex = true;
+
+    /**
+     * HllConfig with empty columnsToDeriveHllFields, default hll log2m, default hll suffix.
+     */
+    public HllConfig() {
+        this(new ArrayList<String>());
+    }
+
+    /**
+     * HllConfig with default hll log2m, default hll suffix.
+     * @param columnsToDeriveHllFields columns to generate hll index
+     */
+    public HllConfig(List<String> columnsToDeriveHllFields) {
+        this(columnsToDeriveHllFields, HllConstants.DEFAULT_LOG2M);
+    }
+
+    /**
+     * HllConfig with default hll suffix.
+     * @param columnsToDeriveHllFields columns to generate hll index
+     * @param hllLog2m The log2m parameter defines the accuracy of the counter.
+     *                 The larger the log2m the better the accuracy.
+     *                 accuracy = 1.04/sqrt(2^log2m)
+     */
+    public HllConfig(List<String> columnsToDeriveHllFields, int hllLog2m) {
+        this(columnsToDeriveHllFields, hllLog2m, HllConstants.DEFAULT_HLL_DERIVE_COLUMN_SUFFIX);
+    }
+
+    /**
+     *
+     * @param columnsToDeriveHllFields columns to generate hll index
+     * @param hllLog2m The log2m parameter defines the accuracy of the counter.
+     *                 The larger the log2m the better the accuracy.
+     *                 accuracy = 1.04/sqrt(2^log2m)
+     * @param hllDeriveColumnSuffix suffix of column used for hll index
+     */
+    public HllConfig(List<String> columnsToDeriveHllFields, int hllLog2m, String hllDeriveColumnSuffix) {
+        Preconditions.checkNotNull(columnsToDeriveHllFields, "ColumnsToDeriveHllFields should not be null.");
+        Preconditions.checkNotNull(hllDeriveColumnSuffix, "HLL Derived Field Suffix should not be null.");
```

Comment:
This has already been checked inside the function of HllUtil.getHllFieldSizeFromLog2m(hllLog2m).